test/Streamly/Test/Data/Array/Stream/Foreign.hs

```diff
@@ -71,6 +72,31 @@ splitOnSuffix sep inp out = do
 moduleName :: String
 moduleName = "Data.Array.Stream.Foreign"
 
+-- Instead of hard coding 10000 here we can have maxStreamLength for operations
+-- that use stream of arrays.
+
+concatArrayW8 :: Property
+concatArrayW8 =
+    forAll (vectorOf 10000 (arbitrary :: Gen Word8))
+        $ \w8List -> do
+              let w8ArrList = Array.fromList . (: []) <$> w8List
+              f2 <- Stream.toList $ ArrayStream.concat $ Stream.fromList w8ArrList
+              w8List `shouldBe` f2
+
+testAppend ::  Property
+testAppend =
+   forAll (listOf (chooseInt (-50, 100))) $ \ls0 ->
+        monadicIO $ action ls0
+
+        where
+
+        action ls = do
+            x <- Stream.fold
+                    (MArray.append (MArray.newArray 0))
+                    (Stream.fromList (ls::[Int]))
+            lst <- MArray.toList x
+            assert (ls == lst)
```

Comment:
This is not an array stream test case. Should not be in this module. This should be in mutable array test module.